docs/or-return-syntax.md

```diff
@@ -0,0 +1,71 @@
+# `or return` Syntax
+
+## Summary
+
+The `or return` syntax is a more compact, understandable way to end execution if an expression evaluates to `nil`.
```

Comment:
Inconsistent with usual truthiness test. What if the expression evaluates to `false`?

Reply:
Probably should have clarified here - the main use case is for expressions that can return `nil`, but the already-established truthiness rules would be respected.